flumes/discoverer.py

```diff
@@ -364,7 +366,28 @@ def on_file_found(self, enum, res, user_data):
     def on_directory_content(self, path, res, user_data):
         enum = path.enumerate_children_finish(res)
         enum.next_files_async(1, 0, None, self.on_file_found, None)
-
+    
+    def on_usr1_signal(self, signum):
+        if self.numdirs or self.numdiscoveries:
+            logger.debug('Currently scanning, nothing to do') 
+            return
+        elif signal.SIGUSR1:
+            self.scan_root()
+            self.signal_received = False
```

Comment:
This signal_received = False is not needed. It has to be set once the full scanning is done and the signal_received = True